packages/core/streams/langchain-stream.ts

```diff
@@ -3,20 +3,77 @@ import { type AIStreamCallbacks, createCallbacksTransformer } from './ai-stream'
 export function LangChainStream(callbacks?: AIStreamCallbacks) {
   const stream = new TransformStream()
   const writer = stream.writable.getWriter()
+
+  const runs = new Set()
+
+  const handleError = async (e: Error, runId: string) => {
+    runs.delete(runId)
+    await writer.ready
+    await writer.abort(e)
```

Comment:
You might want to be careful with this one, in some chains that eg run multiple things in parallel there could be other events arriving after an error event, it doesn't look like you're handling those

Reply:
How should it be handled? If there's an error, can it recover gracefully from it?

Right now this is definitely giving up on first error, but I'll be honest I haven't encountered a scenario where the error was recovered and it could continue the chain.

Reply:
So it will overall be cancelled when the first error occurs. The issue is if it's a chain with multiple LLM calls in parallel you may still receive new tokens from call #2 after call #1 has failed. In which case currently what would happen is it would probably throw an exception in handleLLMNewToken as you try to write to an already aborted writer?

Reply:
Note: the exception of writing to a stream in closed state seems to be handled somewhere that doesn't seem to cause catastrophic errors.

Reply:
Yes any exceptions inside a callback handler are caught by langchain to avoid an error in a callback handler interrupting a run

Reply:
Ok I see your point. 

How does this suggestion sound. 

Keep an error state handler

```javascropt
let errors = []
```

Whenever we encounter an error on any handle*Error push it errors

Then `handleEnd`

```javascript
  const handleEnd = async (runId: string) => {
    runs.delete(runId)

    if (runs.size === 0) {
      if (error.length !=0) {
          await writer.ready
          await writer.abort(e)
          return;
      }
      await writer.ready
      await writer.close()
    }
  }
```

Reply:
Or I can just re-throw the error similar to https://github.com/vercel-labs/ai/blob/main/packages/core/streams/ai-stream.ts#L152-L154

 I think an issue I had yesterday working with this is that even  though the stream had ended, the background processing had not stopped. This led me to have to restart dev server to clear out running processes

Reply:
I'm a fan of rethrowing the error